### PR TITLE
chore: Setup conventional commits (#49)

### LINKED ISSUE #49: Adopt conventional commits in our workflow
I would like this repo to adopt conventional commits as a way to enforce that our changelog is consistent and can be used to automatically generate changelogs when we are release new versions.

https://www.conventionalcommits.org/en/v1.0.0/

A quick overview:

> The commit contains the following structural elements, to communicate intent to the consumers of your library:
> 
> fix: a commit of the type fix patches a bug in your codebase (this correlates with PATCH in semantic versioning).
> feat: a commit of the type feat introduces a new feature to the codebase (this correlates with MINOR in semantic versioning).
> BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type.
> types other than fix: and feat: are allowed, for example @commitlint/config-conventional (based on the the Angular convention) recommends build:, chore:, ci:, docs:, style:, refactor:, perf:, test:, and others.
> footers other than BREAKING CHANGE: <description> may be provided and follow a convention similar to git trailer format.

The benefits of this kind of format is that thanks to its structure, tooling exist around it to automatically generate change logs in a readable and structured form based on the commit messages. See https://www.conventionalcommits.org/en/v1.0.0/#tooling-for-conventional-commits

Also, it is possible to add some github actions like https://github.com/outillage/commitsar directly to the project to make sure that every commit follows the standard.

### PR DESCRIPTION
**Pull request by redwarp**
Setup commitsar as a github action, and prepare for usage of standard-version as future release tool.

Let's check if commitsar can run if I'm creating the PR from a branch on this repo, and not the fork.

### CODE REVIEW

**Comment by redwarp** [comment]
Yep, you can see the action from here.

File: README.md [hunk 1]
```diff
@@ -107,13 +107,33 @@ Repeat as necessary. The project should build from within Android Studio and if
...
+## Making a release
+We switched to using [standard-version](https://github.com/conventional-changelog/standard-version) to handle our releases. standard-version is a utility for versioning using [semver](https://semver.org/) and CHANGELOG generation powered by Conventional Commits.
+
+If you are one of the members and must release a new version, then check its documentation. You will need to install `npm` if you don't have it. (We recommand using `homebrew` to do so.)
+
+With `npm` install, install standard-version globally: `npm i -g standard-version`
+
+From master, run `standard-version`.
+
+`standard-version` will take care of
```
Comment (argo49) [inline README.md]: nit: remove "take care of" so we don't need to change the tense of the verbs in the list

File: README.md [hunk 2]
```diff
@@ -107,13 +107,33 @@ Repeat as necessary. The project should build from within Android Studio and if
...
+* Detect the previous version tag (based on git tags)
+* Infer the next version number, based on previous standard commit messages. A breaking change will trigger a major bump, a new feature will trigger a minor bump, and a fix will trigger a patch bump.
+* Update the CHANGELOG.md file with the commit messages, as well as the [GlobalDependency.kt](buildSrc/src/main/java/com/yelp/gradle/bento/GlobalDependencies.kt) file with the new version number.
+* Create a new commit with these change, and tag it with the version number.
+
+The rest has to be done manually: check that the commit looks good (proper changelog, version number, tag), then push both commit and tags to master.
+
+Publishing to maven is also done manually for now. Check with other members about how to proceed.
```
Comment (argo49) [inline README.md]: Might be a good idea to mention we have an internal wiki page on deploying to maven.

**Review by argo49 — APPROVED** [review]
This is pretty cool!

File: README.md (hunk 2 above)
Comment (redwarp) [reply README.md]: Good point

File: .github/workflows/pull_requests.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+name: Pull request
+on: pull_request
+
+jobs:
+  commitsar:
+    runs-on: ubuntu-latest
```
Comment (dwaxemberg) [inline .github/workflows/pull_requests.yml:6]: Is this safe? Should we pin to an LTS version?

**Review by dwaxemberg — APPROVED** [review]

File: .github/workflows/pull_requests.yml (hunk 3 above)
Comment (redwarp) [reply .github/workflows/pull_requests.yml:6]: @dwaxemberg I checked, and apparently that is how github does action 🤷‍♂

**Pull request merged**